app/assets/javascripts/analytics/ecommerce.js

```diff
@@ -4,15 +4,17 @@
   window.GOVUK = window.GOVUK || {};
 
   var DEFAULT_LIST_TITLE = 'Site search results';
+  var DEFAULT_TRACK_CLICK_LABEL = 'Results';
 
   var Ecommerce = function (config) {
     this.init = function (element) {
       // Limiting to 100 characters to avoid noise from extra longs search queries
       // and to stop the size of the payload going over 8k limit.
-      var searchQuery   = GOVUK.analytics.stripPII(element.attr('data-search-query')).substring(0, 100).toLowerCase();
-      var ecommerceRows = element.find('[data-ecommerce-row]');
-      var startPosition = parseInt(element.data('ecommerce-start-index'), 10);
-      var listTitle     = element.data('list-title') || DEFAULT_LIST_TITLE;
+      var searchQuery     = GOVUK.analytics.stripPII(element.attr('data-search-query')).substring(0, 100).toLowerCase();
+      var ecommerceRows   = element.find('[data-ecommerce-row]');
+      var startPosition   = parseInt(element.data('ecommerce-start-index'), 10);
+      var listTitle       = element.data('list-title') || DEFAULT_LIST_TITLE;
+      var trackClickLabel = element.data('track-click-label') || DEFAULT_TRACK_CLICK_LABEL;
```

Comment:
I think I understand this change but it's a bit confusing. You're adding the ability to specify a different list from `Results` for this event to be included in, but doing so with a new variable called `trackClickLabel` when there's already a variable that does something different called `listTitle`? Does my confusion make sense? Could this be rewritten to be clearer? Or maybe rewrite the PR description?

Reply:
I'll make it clearer with screenshots, but list title is the name of the ecommerce list, whereas this allow for a flexible eventLabel on click, not always "Results"

Reply:
reverted

Reply:
updated PR description